src/test/java/school/hei/haapi/integration/AttendanceIT.java

```diff
@@ -0,0 +1,241 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import school.hei.haapi.SentryConf;
+import school.hei.haapi.endpoint.rest.api.AttendanceApi;
+import school.hei.haapi.endpoint.rest.client.ApiClient;
+import school.hei.haapi.endpoint.rest.client.ApiException;
+import school.hei.haapi.endpoint.rest.model.AttendanceMovementType;
+import school.hei.haapi.endpoint.rest.model.CourseSession;
+import school.hei.haapi.endpoint.rest.model.CreateAttendanceMovement;
+import school.hei.haapi.endpoint.rest.model.PlaceEnum;
+import school.hei.haapi.endpoint.rest.model.Student;
+import school.hei.haapi.endpoint.rest.model.StudentAttendance;
+import school.hei.haapi.endpoint.rest.model.StudentAttendanceMovement;
+import school.hei.haapi.endpoint.rest.security.cognito.CognitoComponent;
+import school.hei.haapi.integration.conf.AbstractContextInitializer;
+import school.hei.haapi.integration.conf.TestUtils;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@Testcontainers
+@ContextConfiguration(initializers = AttendanceIT.ContextInitializer.class)
+@AutoConfigureMockMvc
+class AttendanceIT {
+  @MockBean
+  private SentryConf sentryConf;
+  @MockBean
+  private CognitoComponent cognitoComponent;
+
+  private static ApiClient anApiClient(String token) {
+    return TestUtils.anApiClient(token, ContextInitializer.SERVER_PORT);
+  }
+
+  @BeforeEach
+  void setUp() {
+    setUpCognito(cognitoComponent);
+  }
+
+  @Test
+  void manager_create_attendance_movement_ok() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    AttendanceApi api = new AttendanceApi(manager1Client);
+
+    StudentAttendanceMovement actual = api.createAttendanceMovement(createAttendanceMovement());
+    StudentAttendanceMovement expected = new StudentAttendanceMovement()
+        .id("attendance1_id")
+        .place(PlaceEnum.ANDRAHARO)
+        .createdAt(Instant.parse("2021-11-08T07:30:00.00Z"))
+        .student(student1())
+        .attendanceMovementType(AttendanceMovementType.IN);
+
+    assertEquals(expected.getCreatedAt(), actual.getCreatedAt());
+    assertEquals(expected.getAttendanceMovementType(), actual.getAttendanceMovementType());
+    assertEquals(expected.getPlace(), actual.getPlace());
+    assertEquals(expected.getStudent(), actual.getStudent());
+  }
+
+  @Test
+  void manager_create_attendance_with_no_student_id() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    AttendanceApi api = new AttendanceApi(manager1Client);
+
+    assertThrowsApiException("{\"type\":\"400 BAD_REQUEST\",\"message\":\"the student with #student_id_ko doesn't exist\"}",
```

Comment:
`400` or `404` ? because the resource is not found

Reply:
checked